homeassistant/components/switch/unifi.py

```diff
@@ -0,0 +1,222 @@
+        result = await progress
+        await update_controller()
+        progress = None
+        update_progress.clear()
+        return result
+
+    async def update_controller():
+        """Update the values of the controller."""
+        tasks = [async_update_items(
+            controller, async_add_entities, request_update,
+            switches, switches_off, update_progress, store
+        )]
+        await asyncio.wait(tasks)
+
+    await update_controller()
+
+
+async def async_update_items(controller, async_add_entities,
+                             request_controller_update, switches, switches_off,
+                             progress_waiting, store):
+    """Update POE port state from the controller."""
+    if not controller.available:
+        for client_id, client in switches.items():
+            if client_id not in progress_waiting:
+                client.async_schedule_update_ha_state()
+        return
+
+    new_switches = []
+    devices = controller.api.devices
+    for client_id in controller.api.clients:
+
+        if client_id in progress_waiting:
+            continue
+
+        if client_id in switches:
+            LOGGER.debug("Updating UniFi switch %s (%s)",
+                         switches[client_id].entity_id,
+                         switches[client_id].client.mac)
+            switches[client_id].async_schedule_update_ha_state()
+            continue
+
+        client = controller.api.clients[client_id]
+        if not client.is_wired or client.sw_mac not in devices or \
+           not devices[client.sw_mac].ports[client.sw_port].port_poe or \
+           not devices[client.sw_mac].ports[client.sw_port].poe_enable and \
+           client.mac not in switches_off:
+            continue
+
+        poe_mode = None
+        if client.mac in switches_off:
+            poe_mode = switches_off[client.mac]['poe_mode']
+
+        switches[client_id] = UniFiSwitch(
+            client, controller, request_controller_update, poe_mode)
+        new_switches.append(switches[client_id])
+        LOGGER.debug("New UniFi switch %s (%s)", client.hostname, client.mac)
+
+    @callback
+    def _data_to_save():
+        """Update storage with devices that has got its' POE off."""
+        return switches_off
+
+    for switch in switches.values():
+        if switch.port.poe_mode == 'off' and \
+           switch.client.mac not in switches_off:
+            switches_off[switch.client.mac] = {
+                'data': switch.client.raw,
+                'poe_mode': switch.poe_mode
+            }
+            store.async_delay_save(_data_to_save, SAVE_DELAY)
+
+        elif (switch.port.poe_mode != 'off' and
+              switch.client.mac in switches_off):
+            del switches_off[switch.client.mac]
+            store.async_delay_save(_data_to_save, SAVE_DELAY)
+
+    if new_switches:
+        async_add_entities(new_switches)
+
+
+class UniFiSwitch(SwitchDevice):
```

Comment:
Even if UniFi is the brand consider using the python style naming "UnifiSwitch" as an option